File: onebot/src/main/java/cn/evole/onebot/sdk/event/notice/group/GroupCardChangeNoticeEvent.java
package cn.evole.onebot.sdk.event.notice.group;

import cn.evole.onebot.sdk.event.notice.NoticeEvent;
import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

/**
 * Created on 2022/7/8.
 *
 * @author cnlimiter
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder(toBuilder = true)
@EqualsAndHashCode(callSuper = true)
public class GroupCardChangeNoticeEvent extends NoticeEvent {

    @SerializedName( "card_new")
    public String cardNew;

    @SerializedName( "group_id")
    public long groupId;

    @SerializedName( "card_old")
    public String cardOld;


}

File: onebot/src/main/kotlin/sdk/event/notice/group/GroupCardChangeNoticeEvent.kt
package cn.evolvefield.onebot.sdk.event.notice.group

import cn.evolvefield.onebot.sdk.event.notice.NoticeEvent
import com.google.gson.annotations.SerializedName
import lombok.AllArgsConstructor
import lombok.Data
import lombok.EqualsAndHashCode
import lombok.NoArgsConstructor
import lombok.experimental.SuperBuilder

@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder(toBuilder = true)
@EqualsAndHashCode(callSuper = true)
class GroupCardChangeNoticeEvent : NoticeEvent() {
    @SerializedName("card_new")
    var cardNew = ""
    @SerializedName("card_old")
    var cardOld = ""
    @SerializedName("group_id")
    var groupId = 0L
}
